src/_guides/language/analysis-options.md

```diff
@@ -0,0 +1,268 @@
+[analyzer package](https://pub.dartlang.org/packages/analyzer) docs and the
+[Analysis Server API Specification](https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html).
+
+
+## The analysis options file
+
+The `.analysis_options` (or `.analysis_options.yaml`) file,
+written in the YAML language, is typically placed in the same
+directory as the pubspec file at the top of the package.
+If no file is present, the analyzer walks up the directory
+tree, looking for one.
+
+Here's a sample YAML analysis_options file:
+
+{% prettify yaml %}
+analyzer:
+  strong-mode:
+    - no-implicit-downcasts: true
+    - no-implicit-dynamic: true
+  errors:
+    todo: ignore
+  exclude:
+    - flutter/**
+    - lib/api/*.dart
+
+linter:
+  rules:
+    - avoid_empty_else
+    - cancel_subscriptions
+    - close_sinks
+{% endprettify %}
+
+YAML is sensitive to whitespace&mdash;don't use tabs in a YAML file,
+and use 2 spaces to denote each level of indentation.
+
+<div class="alert alert-info" markdown="1">
+Note: You might come across a `language:` tag in an analysis options file.
+This tag is used for testing experimental features. You can ignore it.
+</div>
+
+## Specifying strong mode
+
+The Dart language spec supports dynamic typing, allowing you to
+write code that has no type annotations at all.
+Strong mode applies more restrictive rules to the type system and,
+as a result, finds more errors during static analysis and at runtime.
+Another benefit of strong mode is faster compilation.
+Some tools, such as DDC, require strong mode compliance.
+
+The simplest way to enable strong mode is to specify
+`strong-mode: true` in the analysis-options file:
+
+{% prettify yaml %}
+analyzer:
+  strong-mode: true
+{% endprettify %}
+
+This flag defaults to false. Instead of specifying `true`
+you can use the following flags to look for specific types
+of implicit casting, on top of the standard strong mode checks.
+The presence of either flag, regardless of value, enables strong mode.
+
+`no-implicit-downcasts: <bool>`
+: A value of `false` ensures that the type inference engine never
+  implicitly casts to a more specific type. The following valid Dart code
+  includes an implicit downcast that would be caught by this flag:
+
+{% prettify dart %}
+Object o = ...;
+String s = o;  // Implicit downcast
+String s2 = s.substring(1);
+{% endprettify %}
+
+  This flag defaults to `true`.
+
+`no-implicit-dynamic: <bool>`
+: A value of `false` ensures that the type inference engine never chooses
+  the `dynamic` type when it can't determine a static type.
+  This flag defaults to `true`.
+
+To enable both flags:
+
+{% prettify yaml %}
+analyzer:
+  strong-mode:
+    - no-implicit-downcasts: false
+    - no-implicit-dynamic: false
+{% endprettify %}
+
+## Enabling linter rules
+
+The analyzer package also provides a code linter.
+A [wide variety](http://dart-lang.github.io/linter/lints/)
```

Comment:
Expand the link to include "linter rules", since "wide variety" by itself doesn't describe the link target.